src/ddprof_worker.cc

```diff
@@ -584,21 +587,21 @@ void ddprof_pr_lost(DDProfContext &ctx, const perf_event_lost *lost,
 }
 
 DDRes ddprof_pr_comm(DDProfContext &ctx, const perf_event_comm *comm,
-                     int watcher_pos) {
+                     int watcher_pos, PerfClock::time_point timestamp) {
   // Change in process name (assuming exec) : clear all associated dso
   if (comm->header.misc & PERF_RECORD_MISC_COMM_EXEC) {
     LG_DBG("<%d>(COMM)%d -> %s", watcher_pos, comm->pid, comm->comm);
-    DDRES_CHECK_FWD(worker_pid_free(ctx, comm->pid));
+    DDRES_CHECK_FWD(worker_pid_free(ctx, comm->pid, timestamp));
   }
   return {};
 }
 
 DDRes ddprof_pr_fork(DDProfContext &ctx, const perf_event_fork *frk,
-                     int watcher_pos) {
+                     int watcher_pos, PerfClock::time_point timestamp) {
   LG_DBG("<%d>(FORK)%d -> %d/%d", watcher_pos, frk->ppid, frk->pid, frk->tid);
   if (frk->ppid != frk->pid) {
     // Clear everything and populate at next error or with coming samples
-    DDRES_CHECK_FWD(worker_pid_free(ctx, frk->pid));
+    DDRES_CHECK_FWD(worker_pid_free(ctx, frk->pid, timestamp));
     ctx.worker_ctx.us->dso_hdr.pid_fork(frk->pid, frk->ppid);
```

Comment:
It would be better to only call `pid_fork` when pid mappings have actually been freed.